packages/core/src/serializerUtils.ts

```diff
@@ -27,7 +27,7 @@ export function serializeNumber(value: number | bigint): Uint8Array {
     } else {
       const buffer = new ArrayBuffer(8);
       const view = new DataView(buffer);
-      view.setBigUint64(0, value as bigint);
+      view.setBigInt64(0, BigInt(value));
```

Comment:
confusingly, data views need to set a value as "BigInt" for 64 bit numbers that would otherwise be integers.